src/ape_test/providers.py

```diff
@@ -57,6 +57,13 @@ def update_settings(self, new_settings: dict):
         pass
 
     def estimate_gas_cost(self, txn: TransactionAPI, **kwargs) -> int:
+        if isinstance(self.network.gas_limit, int):
+            return self.network.gas_limit
+
+        if self.network.gas_limit == "max":
```

Comment:
This is really a nit but `elif` is good in case things change, we remember that the first `if` should really not fallthrough

```suggestion
        elif self.network.gas_limit == "max":
```